mesonpy/_tags.py

```diff
@@ -1,130 +1,124 @@
-
-
-class StableABITag(Tag):
-    _REGEX = re.compile(r'^abi(?P<abi_number>[0-9]+)$')
-
-    def __init__(self, value: str) -> None:
-        match = self._REGEX.match(value)
-        if not match:
-            raise ValueError(f'Invalid PEP 3149 stable ABI tag, expecting pattern `{self._REGEX.pattern}`')
-        self._abi_number = int(match.group('abi_number'))
-
-    @property
-    def abi_number(self) -> int:
-        return self._abi_number
-
-    def __str__(self) -> str:
-        return f'abi{self.abi_number}'
-
-    @property
-    def python(self) -> Literal[None]:
-        return None
-
-    @property
-    def abi(self) -> str:
-        return f'abi{self.abi_number}'
-
-    def __eq__(self, other: Any) -> bool:
-        return isinstance(other, self.__class__) and other.abi_number == self.abi_number
-
-    def __hash__(self) -> int:
-        return hash(str(self))
-
-
-class InterpreterTag(Tag):
-    def __init__(self, value: str) -> None:
-        parts = value.split('-')
-        if len(parts) < 2:
-            raise ValueError(
-                'Invalid PEP 3149 interpreter tag, expected at '
-                f'least 2 parts but got {len(parts)}'
-            )
-
-        # On Windows, file extensions look like `.cp311-win_amd64.pyd`, so the
-        # implementation part (`cpython-`) is different from Linux. Handle that here:
-        if parts[0].startswith('cp3'):
-            parts.insert(0, 'cpython')
-            parts[1] = parts[1][2:]  # strip 'cp'
-
-        self._implementation = parts[0]
-        self._interpreter_version = parts[1]
-        self._additional_information = parts[2:]
-
-        if self.implementation != 'cpython' and not self.implementation.startswith('pypy'):
-            raise NotImplementedError(f'Unknown Python implementation: {self.implementation}.')
-
-    @property
-    def implementation(self) -> str:
-        return self._implementation
-
-    @property
-    def interpreter_version(self) -> str:
-        return self._interpreter_version
-
-    @property
-    def additional_information(self) -> Sequence[str]:
-        return tuple(self._additional_information)
+import os
+import platform
+import sys
+import sysconfig
+
+from typing import Optional, Union
+
+
+# https://peps.python.org/pep-0425/#python-tag
+INTERPRETERS = {
+    'python': 'py',
+    'cpython': 'cp',
+    'pypy': 'pp',
+    'ironpython': 'ip',
+    'jython': 'jy',
+}
+
+
+def get_interpreter_tag() -> str:
+    name = sys.implementation.name
+    name = INTERPRETERS.get(name, name)
+    version = sys.version_info
+    return f'{name}{version[0]}{version[1]}'
+
+
+def _get_config_var(name: str, default: Union[str, int, None] = None) -> Union[str, int, None]:
+    value = sysconfig.get_config_var(name)
+    if value is None:
+        return default
+    return value
+
+
+def _get_cpython_abi() -> str:
+    version = sys.version_info
+    debug = pymalloc = ''
+    if _get_config_var('Py_DEBUG', hasattr(sys, 'gettotalrefcount')):
```

Comment:
The idea is to rely on the value of `Py_DEBUG` if set (`sysconfig.get_config_var()` does not return Null) and fallback to the `hasattr` check otherwise. Thus open-coding this would be something like:
```
debug = _get_config_var('Py_DEBUG')
if debug is None:
    debug = hasattr(sys, 'gettotalrefcount')
```
Because we have two of these checks I decided to have the tiny helper function. Happy to open code it if you prefer.